### PR TITLE
Fix cflags for unix

### PR DESCRIPTION
**Pull request by Thrameos**
This is a shot at fixing the distribution size for unix.   I changed the patterns so we can explicitly set the debug and optimization flags from platform.  The requires scrubbing all the existing flags in the sysconfig.

Also fixes a bug with coverage and tracing.   Not sure why the extension position is hard coded, but the number drifted when we added the jar to the extension list.

### CODE REVIEW

File: setupext/build_ext.py [hunk 1]
```diff
@@ -203,40 +191,41 @@ def initialize_options(self, *args):
...
+            for r in remove_args:
+                args = list(filter((r).__ne__, args))
+
+            cfg_vars[k] = " ".join(args)
         super().initialize_options()
 
     def _set_cflags(self):
         # set compiler flags
         c = self.compiler.compiler_type
+        jpypeLib = self.extensions[1]
```
Comment (marscher) [inline setupext/build_ext.py:214]: this static indexing will hurt us in the future, won't it? (it already did, eventually put an assertion to match the desired name at least).

**Review by marscher — APPROVED** [review]

**Comment by marscher** [comment]
Should we consider using an newer c++ standard, eg. 17? I think on Windows we already use it

File: setupext/build_ext.py (hunk 1 above)
Comment (Thrameos) [reply setupext/build_ext.py]: I address this with a list comprehension.   Should be a bit more future proof.

**Comment by Thrameos** [comment]
It is true that standard 11 is rather ancient.  We can play around with other standards at some point.   Not sure what is the oldest compiler that we are likely to encounter currently.   The only reason for the VC to use a later standard is that the nothrow logic didn't get turned on without it.   As we are not currently using any later C++ other features I am not seeing much in the way of compelling reasons to change.   My time is a bit limited today (even through it is listed as my scheduled day off), so I am going to just try to cut a patch release.

**Pull request merged**